lms/djangoapps/certificates/models.py

```diff
@@ -890,10 +891,13 @@ def set_enabled_for_course(cls, course_key, is_enabled):
             is_enabled (boolean): Whether to enable or disable self-generated certificates.
 
         """
-        CertificateGenerationCourseSetting.objects.create(
-            course_key=course_key,
-            enabled=is_enabled
-        )
+        certificate_generation_course_setting = CertificateGenerationCourseSetting(course_key=course_key,
```

Comment:
There is really no need to write an atomic block for this as `CertificateGenerationCourseSetting.objects.create` will be done in atomic transaction by default: https://github.com/django/django/blob/master/django/db/models/query.py#L373-L381

Also there is no need to catch integrity errors.